mempool/metrics.go

```diff
@@ -72,6 +72,7 @@ func PrometheusMetrics(namespace string, labelsAndValues ...string) *Metrics {
 	for i := 0; i < len(labelsAndValues); i += 2 {
 		labels = append(labels, labelsAndValues[i])
 	}
+	typedCounterLabels := append(append(make([]string, 0, len(labels)+1), labels...), TypeLabel)
```

Comment:
why not just append the `TypeLabel`?

Reply:
we probably could since I don't think we're writing to it later, but I've ran into too many difficult to find slice bugs in my life.

Since we're not overwriting an existing var, my fingers literally won't let me use append using an existing slice.